lib/market.rb

```diff
@@ -0,0 +1,75 @@
+require 'date'
+class Market
+    attr_reader :name, :vendors
+    def initialize(name)
+        @name = name
+        @vendors = []
+        @date = Date.today
+        #require 'pry'; binding.pry
+    end
+    
+    def date
+        @date.strftime("%d/%m/%Y")
+    end
+
+    def add_vendor(vendor)
+        @vendors << vendor
+    end
+
+    def vendor_names
+        names = []
+        @vendors.each do |vendor|
+            names << vendor.name
+        end
+        names
+    end
+
+    def vendors_that_sell(item)
+        results = []
+        @vendors.each do |vendor|
+            if vendor.check_stock(item) > 0
+                results << vendor
+            end
+        end
+        results
+    end
+
+    def total_inventory
```

Comment:
Try to do this with reduce instead - its generally a better enumerable if youre building a hash.